rootgrow/rootgrow.py

```diff
@@ -70,36 +71,12 @@ def get_disk_device_from_root(root_device):
                 msg=err_msg
             )
         )
-    for device_info in stdout_data.decode().split(os.linesep):
-        device_list = device_info.split()
-        if device_list and device_list[1] == 'disk':
-            return device_list[0]
+    return stdout_data.strip().decode()
 
 
-def get_partition_id_from_root(disk_device, root_device):
-    proc = subprocess.Popen(
-        ['lsblk', '-p', '-n', '-r', '-o', 'NAME', disk_device],
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE
-    )
-    stdout_data, stderr_data = proc.communicate()
-    err_msg = stderr_data.decode()
-    if err_msg:
-        raise Exception(
-            'Unable to determine root partition index {}'.format(
-                err_msg
-            )
-        )
-    partition_index = ''
-    for device in stdout_data.decode().split(os.linesep):
-        if device:
-            if device == root_device:
-                for c in reversed(device):
-                    if c.isdigit():
-                        partition_index = c + partition_index
-                    else:
-                        return int(partition_index)
-    raise Exception('Unable to determine root partition index')
+def get_partition_id_from_root(root_device):
+    partition_index = re.match('.*?([0-9]+)$', root_device).group(1)
```

Comment:
I once had a problem and used a regular expression to solve it, then I had 2 problems ;)

Anyway, I think we should match the beginning of the string, i.e. start with '^' then the combination of '.*?' is not necessarily intuitive and probably deserves a comment such as 

`Use ? to make matching less greedy`

'*' -> Zero or more occurrences of the preceding regular expression, which in this case is '.', i.e. any character
'?' -> Zero or one occurrence  of the preceding regular expression, which in this case is '.*', i.e. any character combination

Then the end of line match is outside the group but we have to make sure that the numbers get matched at the end of the string as such '$' belongs inside the group, IMHO.

Last but not least I'd replace the range with the pattern sequence, i.e. '\d'

I think the regular expression should look as follows:

`^.*?(\d+$)`

With the appropriate comment as pointed out above.

Reply:
IMHO the end of line match makes more sense outside of the group since the group should really just be the partition number. However, it works the same either way. The beginning of line match also doesn't matter it works the same with or without it. Technically we could simply use something like `(\d+)$`. But I'm okay with the suggestion or preferably `^.*?(\d+)$`.

Reply:
Yes for our test cases the results are all the same. You are correct `$` should eb outside we don't want that match inside the group we only want all the numbers from the end of the string, so `^.*?(\d+)$` it is. And please add the comment.

Thanks